statsmodels/tsa/statespace/innovation.py

```diff
@@ -0,0 +1,72 @@
+from statsmodels.tsa.statespace.mlemodel import MLEModel, MLEResults, MLEResultsWrapper
+
+
+class InnnovationModel(MLEModel):
+    def __init__(self, endog, k_states, **kwargs):
+        kwargs.update(k_posdef=1)
+        k_states += 1
+        super().__init__(endog, k_states, **kwargs)
+        self.ssm['design', 0, -1] = 1
+        self.ssm['selection', -1, 0] = 1
+
+    def __setitem__(self, key, value):
+
+        _type = type(key)
+        if _type is str:
+            if key == 'cov':
+                self.ssm['state_cov', 0, 0] = value
+            elif key == 'transition':
+                self.ssm['transition', :-1, :-1] = value
+            elif key == 'selection':
+                self.ssm['transition', :-1, -1:] = value
```

Comment:
[http://nbviewer.jupyter.org/gist/ChadFulton/f0b8fc2e7d47a0103f97](http://nbviewer.jupyter.org/gist/ChadFulton/f0b8fc2e7d47a0103f97)
and the origin google groups discuss is here
[confuse with the example of AR(1) State Space Model](https://groups.google.com/forum/?hl=zh-Cn#!searchin/pystatsmodels/state$20space$20single/pystatsmodels/zIFQVW82ACU/Gf2xqAtaBgAJ)